### ISSUE TITLE
`.js` files are parsed as CommonJS scripts — JSX in `.js` silently drops the whole file

### ISSUE THREAD

**Issue by rboudrouss** [rboudrouss/reactant-analyzer#87]
## Symptom

The chakra-ui and excalidraw corpus reports each carry 3 `parse_errors[]` entries. All 6 files are valid JavaScript:

| file | message | actual construct | outcome |
|---|---|---|---|
| chakra-ui `sandbox/{iframe,panda-preset,shadow-dom}/eslint.config.js` | Unexpected import.meta expression | valid ESM `import.meta.url` | recovered, analyzed — noise entry |
| excalidraw `dev-docs/src/components/Highlight.js`, `Homepage/index.js`, `theme/ReactLiveScope/index.js` | Unexpected JSX expression | real React components, JSX in `.js` (Docusaurus convention) | **parser panics, file silently skipped** |

The three excalidraw files contain React components (one calls `useColorMode` inside `forwardRef`) that never enter the analysis — silent false negatives, which the project forbids.

## Verified FN repro

The same component analyzed under both extensions:

```jsx
// Loop.jsx / Loop2.js — identical content
import { useState } from 'react';
export function Loop() {
  const [n, setN] = useState(0);
  setN(n + 1);
  return <div>{n}</div>;
}
```

- `reactant check Loop.jsx` → `files_analyzed: 1`, **`setter-in-render` error**
- `reactant check Loop2.js` → `files_analyzed: 0`, **zero diagnostics**, one `parse_errors[]` entry

## Cause

`src/resolver/mod.rs` maps extensions to source types with `_ => SourceType::cjs()`, so every `.js` the walker collects is parsed as a CommonJS **script**: JSX off, `import.meta` illegal. The React ecosystem routinely puts JSX in `.js` (babel, CRA, Docusaurus); oxc's own `SourceType::from_path` treats `.js` as a JSX-enabled module for exactly that reason. Commit 7d1957c (skip only on parser panic) doesn't help: JSX in script mode *panics*, so the file is still dropped.

Surfacing gap on top: `--format json` prints nothing about the skip to stderr, and `parse_errors[]` doesn't distinguish "recovered, analyzed" from "panicked, skipped".

## Fix direction

Parse `.js` as a JSX-enabled module (align with `SourceType::from_path`), and make `parse_errors[]` say whether the file was analyzed or dropped.

**Comment by rboudrouss** [comment]
Corrigé. Le repro FN de l'issue passe maintenant : `Loop2.js` → `files_analyzed: 1`, `setter-in-render` remonté.

## Une correction sur la cause

L'issue dit que `SourceType::from_path` de oxc traite `.js` comme un module JSX. Ce n'est **pas** le cas en oxc 0.138 : `impl From<FileExtension> for SourceType` donne `LanguageVariant::Standard` pour `Js` (seuls `Jsx`/`Tsx` sont `LanguageVariant::Jsx`). S'aligner sur `from_path` n'aurait donc pas suffi — il faut activer JSX explicitement.

## Ce qui a été fait

Une seule fonction partagée, `resolver::source_type_for` :

- `.js` / `.jsx` / tout le reste → `SourceType::unambiguous().with_jsx(true)`. `unambiguous` plutôt que `mjs`/`cjs` pour que le même `.js` supporte `import.meta` **et** `require`/`module.exports` — les deux moitiés du problème sont couvertes par des tests.
- `.ts` / `.mts` / `.cts` → `SourceType::ts()`, inchangé : `<T>expr` est une assertion de type en `.ts` et un élément JSX en `.tsx`, les deux ne peuvent pas partager le flag. Test dédié.

Elle est `pub` : `tests/page_collision.rs` et `tests/multi_file_discovery.rs` gardaient chacun une copie privée de ce `match` (donc du bug). Elles appellent la fonction partagée maintenant.

## Le trou de signalement

Traité aussi, puisque c'est ce qui rendait le FN invisible :

- `ParseError { file, message, analyzed }` remplace le tuple. `analyzed: false` = le fichier est sorti de l'analyse, ses findings sont *manquants*, pas *absents*.
- `parse_errors[].analyzed` exposé en JSON (ajout additif, le schéma reste v2 ; `docs/usage.md` et le REFERENCE du skill de triage sont à jour).
- Un fichier *droppé* est signalé sur stderr **quel que soit le `--format`** — stderr est un flux séparé, stdout reste un unique document JSON. Un fichier simplement récupéré reste sur le canal humain.

## Vérification corpus

Les deux symptômes décrits dans l'issue ont disparu :

| corpus | avant | après |
|---|---|---|
| excalidraw | 3 `parse_errors`, 3 composants React jamais analysés | **0**, 517 fichiers / 263 composants analysés |
| chakra-ui | 3 `parse_errors` (`import.meta`) | **0**, 2626 fichiers analysés |
